lib/net.js

```diff
@@ -1577,19 +1576,13 @@ Server.prototype.getConnections = function(cb) {
 
 
 Server.prototype.close = function(cb) {
-  if (typeof cb === 'function') {
-    if (!this._handle) {
-      this.once('close', function close() {
-        cb(new ERR_SERVER_NOT_RUNNING());
-      });
-    } else {
-      this.once('close', cb);
-    }
-  }
-
   if (this._handle) {
     this._handle.close();
     this._handle = null;
+
+    if (typeof cb === 'function') {
+      this.once('close', cb);
+    }
```

Comment:
@mcollina: This change means that the callback won't always be invoked. Isn't that considered an anti pattern?

Reply:
As long as callback is a function, it will be called. Or am I missing something?

Reply:
@ronag is right. If there is no `_handle` yet the callback will not be called.